packages/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -440,18 +450,21 @@ export const GridRootStyles = styled('div', {
 
     /* Cell styles */
     [`& .${c.cell}`]: {
-      display: 'flex',
-      alignItems: 'center',
```

Comment:
To make things work for string cells without a wrapper, cells need to not be `display: flex` by default (because it messes with `text-overflow: ellipsis`). This breaks the positioning of every column that relied on `align-items: center`, in other words most custom cells. To make the transition as painless as possible, here are the options we have:
 1. Export `gridClasses['cell--flex']` that restores those styles. Users can use `cellClassName: gridClasses['cell--flex']` in their column definition.
 1. Add something like a `display: 'flex'` option on the `GridColDef` itself to make even more convenient to restore the flex behavior
 2. Go the opposite way and make `display: flex` the default, and adjust the styles of every string column with one of the options above.

Reply:
Went for `display: 'flex'`, it seemed like the most convenient option.

Reply:
I don't love this breaking change, can we avoid it?
Did you consider using `::before`? https://codesandbox.io/p/sandbox/bitter-thunder-9w47cl?file=%2Findex.html%3A19%2C21
Not sure if it's any better than 2 DOM nodes performance-wise though.
What do you think?

Reply:
Performance-wise there is no difference between `::before` and a DOM node, in both cases the browser will create a layout object at that location.

Maybe instead we could check if the cell child is a string or an object, and change the display based on that?

Reply:
> Maybe instead we could check if the cell child is a string or an object, and change the display based on that?

Yes, something like this:
```tsx
if (typeof children !== 'string' && typeof children !== 'number') {
  classNames.push(gridClasses['cell--flex']);
}
```

The downside is that the text won't be truncated in this case:

```tsx
renderCell: ({ value }) => <div>{value}</div>
```

Did the removal of the `.cellContent` element improve scrolling performance? Do you have any numbers?

Reply:
I don't love the runtime check for string/object because it feels a bit obscure from a user POV that it would do that, but I'm ok with it. What do you think about the `display` field on the columns? Should we keep it for users to adjust that easily?

> Did the removal of the .cellContent element improve scrolling performance? Do you have any numbers?

Yes, we got a 5% improvement over scripting & rendering. It's not a huge improvement but I think it's worth it. If we keep accumulating the improvements, we'll be more performant than the competition (ag-grid). I think with all of the recent ones and #12023 we might be there soon, I'll need to benchmark.

| Before | After |
| --- | --- |
| ![Screenshot from 2024-02-12 19-28-58](https://github.com/mui/mui-x/assets/1423607/8035ac2e-b292-4ff4-83b7-b3699738dcbc) | ![Screenshot from 2024-02-12 19-28-08](https://github.com/mui/mui-x/assets/1423607/6bbf0ddf-6750-4707-beeb-9f5c0f429a86) |

*Test: scroll a large list with `scroller.scrollTo({ top: 100_000, behavior: 'smooth' })`*

Reply:
To be fair, AG Grid is rarely the fastest in the benchmark https://dzone.com/articles/data-grid-performance-comparison, https://medium.com/@david.greene_40229/the-fastest-javascript-data-grid-a-performance-analysis-d7a34389593e. I suspect they differentiate with the depth of the features available.

Now, I think we should be centered on what do our users want, I'm all for a slick UX 😁

Reply:
Good to know, though the benchmarks in those links seem outdated and don't match what I feel ag-grid current performance is. It would be neat to create benchmarks to compare with other grids. Then we'll be able to write our own article comparing performance where we claim we're the fastest.

Reply:
A blog post on all the techniques and learnings we used to make it fast sounds 👌

Reply:
I was also thinking about reworking our [docs demos](https://mui.com/x/react-data-grid/demo/). In the current state, it feels a bit cramped and unintuitive to use, when compared with e.g. the [ag-grid demo](https://ag-grid.com/example/):

| DataGrid | AG-grid |
| --- | --- |
| ![image](https://github.com/mui/mui-x/assets/1423607/6b2f05cc-92f0-4055-bd3e-396056377ca8) | ![image](https://github.com/mui/mui-x/assets/1423607/53b0e973-e129-436e-ae87-43638130c8cb) |

Maybe even making a similar dataset to what ag-grid uses, so people can do a 1-1 comparison. Grid performance in particular is not really displayed in the docs, we only have small datasets examples. I've been assuming ag-grid is the main competition btw. We should also probably create a feature-by-feature comparison to other grids, with the latest changes we are going to be more performant and more accessible to screen readers than other grids, it would be nice to highlight.

Reply:
IMHO, the added `GridColDef['display']` makes it harder to reuse the custom cell renderer:

```tsx
import { renderEmail } from './grid-utils';

const columns = [
  {
    field: 'email',
    renderCell: renderEmail,
    display: 'flex',
  }
]
```

This could be solved by extracting a custom column though:

```tsx
import { emailColumn } from './grid-utils';

const columns = [
  {
    field: 'email',
    ...emailColumn, // already has `display: 'flex'` applied
  }
]
```

I think it's OK